gems/activerecord/6.0/activerecord.rbs

```diff
@@ -680,8 +682,9 @@ interface _ActiveRecord_Relation[Model, PrimaryKey] # rubocop:disable RBS/Lint/T
                | (?batch_size: Integer, ?start: Integer, ?finish: Integer, ?error_on_ignore: bool) -> Enumerator[Model, nil]
   def find_in_batches: (?batch_size: Integer, ?start: Integer, ?finish: Integer, ?error_on_ignore: bool) { (Array[Model]) -> void } -> nil
   def in_batches: (?of: Integer, ?start: Integer, ?finish: Integer, ?load: bool, ?error_on_ignore: bool, ?order: untyped) { (self) -> void } -> nil
-  def sum: (?untyped? column_name) -> Integer
-         | () { (Model) -> Integer } -> Integer
+  def sum: (?untyped? column_name) -> untyped
+         | [T] () { (Model) -> T } -> (Integer | T)
+         | [U] (U initial_value) { (Model) -> untyped } -> U
```

Comment:
The return value does not necessarily match `initial_value`.

```rb
User.sum(0) { |user| user.decimal_column } #=> BigDecimal
```

Also, if there are zero records, the type of initial_value seems to be returned.

```rb
User.sum(0.0) { |user| user.decimal_column } #=> 0.0
```

If you value practicality,

```suggestion
         | [T] (untyped initial_value) { (Model) -> T } -> T
```

If you value accuracy,

```suggestion
         | [T, U] (T initial_value) { (Model) -> U } -> (T | U)
```